apisix/admin/stream_routes.lua

```diff
@@ -142,6 +165,15 @@ function _M.delete(id)
         return 400, {error_msg = "missing stream route id"}
     end
 
+    local status, body = _M.get()
```

Comment:
Would be better to use synced data from https://github.com/apache/apisix/blob/2f4a4dba2c0cbc426ae99133c2546112ea71dba8/apisix/stream/router/ip_port.lua#L189.

It may be too expensive to read the whole stream routes from etcd each time we want to delete one.

It might be data race but I think it is acceptable. The current implementation still has a smaller data race since we don't use a transaction to deal with the deletion.

Reply:
I have tried to use synced data from  _M.routes() and  it  needs to apisix.stream_init_worker()  before calling .
I agree with you that it is more efficient to   use synced data from _M.routes() , but I  get a lot of errors  while adding  apisix.stream_init_worker()   in stream  subsystem of nginx.conf  
So after that I took the etcd implementation  based on this suggestion  [https://github.com/apache/apisix/pull/8219#issuecomment-1311464368](url).